samples/java-customer-registry/src/main/java/customer/CustomerValueEntityHandler.java

```diff
@@ -28,70 +28,59 @@
 import com.google.protobuf.InvalidProtocolBufferException;
 import customer.api.CustomerApi;
 import customer.domain.CustomerDomain;
-import scala.None$;
-import scala.Option;
 import scalapb.UnknownFieldSet;
+import java.util.Optional;
 
 /**
  * This class would be generated by codegen and should be treated mostly as an implementation detail
  * by the user.
  */
 public class CustomerValueEntityHandler implements ValueEntityHandler {
+
   final CustomerValueEntity entity;
-  final AnySupport anySupport;
 
   public static final Descriptors.ServiceDescriptor serviceDescriptor =
       CustomerApi.getDescriptor().findServiceByName("CustomerService");
   public static final String entityType = "customers";
 
-  CustomerValueEntityHandler(CustomerValueEntity entity) {
-    this.entity = entity;
-    this.anySupport =
-        new AnySupport(
-            new Descriptors.FileDescriptor[] {CustomerDomain.getDescriptor()},
-            CustomerValueEntityHandler.class.getClassLoader(),
-            AnySupport.DefaultTypeUrlPrefix(),
-            AnySupport.PREFER_JAVA());
+  CustomerValueEntityHandler() {
+    this.entity = new CustomerValueEntity();
```

Comment:
I'm not sure we should construct the user implementation from the generated code: that would mean the user no longer has the opportunity to pass in additional utilities that might be needed in their implementation methods

Reply:
Perhaps we can provide some more utilities there, like a registration component that accepts a number of factories/lambdas to create the entities - but let's take one thing at a time ;)

Reply:
@franciscolopezsancho, this is something that affects the testkit. In #149 you asked if the testkit should init the entity by itself or not. My argument was that it was convenient to do so, but Arnout has an import point here. Users may want to define their constructor the way they want and pass things to it. 

So, both handler and testkit should receive an entity as parameter.